crypto/crypto.go

```diff
@@ -20,8 +20,30 @@ func GetPubKeyHex(privateKey PrivateKey) string {
 	return hex.EncodeToString(privKey.PublicKey.X) + hex.EncodeToString(privKey.PublicKey.Y)
 }
 
-func GeneratePrivateKey() (PrivateKey, error) {
-	return elliptic.GeneratePrivateKey(elliptic.Secp256k1)
+func GeneratePrivateKey(difficulty int) (PrivateKey, error) {
+	for true {
+		privKey, err := elliptic.GeneratePrivateKey(elliptic.Secp256k1)
+		pubKey := sha256.Sum256(append(privKey.PublicKey.X, privKey.PublicKey.X...))
+
+		nullBytes := difficulty / 8
+		remainder := difficulty % 8
+		isDifficult := true
+
+		for i := 0; i < nullBytes; i++ {
+			if pubKey[i] != 0 {
+				isDifficult = false
+				break
+			}
+		}
+
+		if isDifficult == true {
+			if pubKey[nullBytes] <= (2<<(8-uint(remainder)-1))-1 {
```

Comment:
it's the same as
`if pubKey[nullBytes] < (1<<(8-uint(remainder))) {`

also you may do `uint` cast at the declaration to avoid it here to make it more readable

Reply:
Nice simplification